### PR TITLE
Feat_google_auth:Added the google auth and fixed the apollo token issue

### PR DESCRIPTION
**Pull request by Amartyakumar**

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -6,69 +6,125 @@ const app = express()
...
 app.post('/connect/token', async (req, res) => {
   const params = new URLSearchParams()
+  const urls = getTokenUserInfoUrl(req.query)
 
   if (req.body.refresh_token) {
     const refreshToken = req.body.refresh_token
     params.append('grant_type', 'refresh_token')
     params.append('refresh_token', refreshToken)
-    params.append('client_id', nuxtconfig.auth.strategies.bitpod.clientId)
+    params.append(
```
Comment (SukantGujar) [inline api/index.js:39]: This entire code is duplicated below. Simplify it.

File: api/index.js [hunk 2]
```diff
@@ -6,69 +6,125 @@ const app = express()
...
       Authorization: req.headers.authorization,
     },
   }
   try {
-    const userResponse = await axios.get(userInfoEndPointUrl, config)
-    res.json({
-      status: 200,
-      data: userResponse.data,
-    })
+    if (req.query.provider === 'google') {
```
Comment (SukantGujar) [inline api/index.js:111]: Again a good amount of repeated code here in the if-else. Simplify.

File: layouts/default.vue [hunk 3]
```diff
@@ -456,10 +451,21 @@ export default {
...
-    method() {
-      // Perform an action
+    async onLogout() {
+      this.$auth.logout()
+      await this.$apolloHelpers.onLogout()
     },
   },
+  async created() {
+    if (!this.$apolloHelpers.getToken()) {
+      debugger
```
Comment (SukantGujar) [inline layouts/default.vue:461]: Why? Please always check for eslint errors before checking in code.

File: api/authorize.js [hunk 4]
```diff
@@ -8,15 +8,16 @@ export default function (req, res, next) {
   }
   // eslint-disable-next-line node/no-deprecated-api
   const host = url.parse(referer).host
-  const query = qs.parse(req.url.substring(2))
+  const query = qs.parse(req.url.substring(2).replace('?', '&'))
```
[the remark this answers is not in the document]
Comment (Amartyakumar) [reply api/authorize.js:11]: Resolved

File: api/index.js [hunk 5]
```diff
@@ -6,69 +6,125 @@ const app = express()
...
+    case 'google':
+      tokenUrl = nuxtconfig.auth.strategies.google.tokenEndPointUrl
+      userInfoUrl = nuxtconfig.auth.strategies.google.userInfoEndPointUrl
+      break
+    case 'bitpod':
+      tokenUrl = nuxtconfig.auth.strategies.bitpod.tokenEndPointUrl
+      userInfoUrl = nuxtconfig.auth.strategies.bitpod.userInfoEndPointUrl
+      break
+    default:
+      console.log('Sorry no provider added')
```
[the remark this answers is not in the document]
Comment (Amartyakumar) [reply api/index.js:22]: Resolved

File: layouts/default.vue (hunk 3 above)
Comment (Amartyakumar) [reply layouts/default.vue:461]: Resolved

**Review by SukantGujar — APPROVED** [review]

**Pull request merged**